install_config/router/default_haproxy_router.adoc

```diff
@@ -1457,6 +1457,129 @@ access any pod in the cluster. If isolation is needed in this case, then do not
 add routes across the namespaces.
 ====
 
+[[using-the-dynamic-configuration-manager]]
+== Using the Dynamic Configuration Manager
+
+You can configure the HAProxy router to support the dynamic configuration
+manager.
+
+The dynamic configuration manager brings certain
+types of routes online without requiring any HAProxy reload downtime. It handles any route and endpoint life
+cycle events such as route and endpoint `addition|deletion|update`.
+
+Enable the dynamic configuration manager by setting the
+`ROUTER_HAPROXY_CONFIG_MANAGER` environment variable to `true`:
+
+----
+$ oc set env dc/<router_name> ROUTER_HAPROXY_CONFIG_MANAGER='true'
+----
+
+If the dynamic configuration manager is unable to dynamically configure HAProxy,
+it forces a configuration re-write and a HAProxy process reload. For example, if
+a newly added route contains custom annotations that would cause a drift from
+the standard configuration (custom timeouts) or if the route requires custom TLS
+configuration.
+
+The dynamic configuration internally uses the haproxy socket and configuration
+API with a pool of pre-allocated routes and back end servers. The pre-allocated
+pool of routes is created using route blueprints. The default set of blueprints
+supports unsecured routes, edge secured routes (without any custom TLS
+configuration), and passthrough routes.
+
+[IMPORTANT]
+====
+`re-encypt` routes require custom TLS configuration information and so are not
+supported out of the box.
+
+However, the blueprints that the dynamic configuration manager uses can be
+extended by setting the `ROUTER_BLUEPRINT_ROUTE_NAMESPACE` and optionally the
+`ROUTER_BLUEPRINT_ROUTE_LABELS` environment variable values. All routes (or the
+routes matching the route labels) in the blueprint route namespace will be
+processed as custom blueprints similar to the default set of blueprints. This
+enables defining certain types of routes such as `re-encypt` routes or routes
+that use custom annotations or routes with custom TLS configuration as
+blueprints.
+====
+
+The following procedure assumes you create three routes: `reencrypt-blueprint`, `annotated-edge-blueprint`, and `annotated-unsecured-blueprint`.
+
+.Procedure
+
+. Create a new project:
++
+----
+$ oc new-project namespace_name
+----
+
+. Create a new route:
++
+----
+$ oc create route edge edge_route_name --key=/path/to/key.pem \
+      --cert=/path/to/cert.pem --service=<service> --port=8443
+----
+
+. Label the route:
++
+----
+$ oc label route edge_route_name type=route_label_1
+----
+
+. xref:../../dev_guide/routes.adoc#creating-routes[Create three different routes]. All have the label `type=route_label_1`:
++
+----
+$ oc create -f reencrypt-blueprint.yaml
+$ oc create -f annotated-edge-blueprint.yaml
+$ oc create -f annotated-unsecured-blueprint.json
+----
++
+You can also remove a label from a route, which prevents it from being used as a
```

Comment:
Is this true even if you don't have a the `ROUTER_BLUEPRINT_ROUTE_LABELS` value set?

Reply:
No, its only true if the `ROUTER_BLUEPRINT_ROUTE_LABELS` is set.